crates/edr_eth/src/signature.rs

```diff
@@ -27,6 +27,9 @@ pub enum SignatureError {
     /// Invalid secret key.
     #[cfg_attr(feature = "std", error("Expected 32 byte secret key"))]
     InvalidSecretKeyLength,
+    /// When parsing a secret key from string to hex
+    #[cfg_attr(feature = "std", error("Invalid hex"))]
+    InvalidSecretKeyHex,
     /// When parsing a signature from string to hex
     #[cfg_attr(feature = "std", error(transparent))]
     DecodingError(#[cfg_attr(feature = "std", from)] hex::FromHexError),
```

Comment:
Should this enum variant be removed now?

Reply:
I pushed some changes after submitting the PR, so not sure if you're referring to this line:

>     /// When parsing a secret key from string to hex
>     #[cfg_attr(feature = "std", error("Invalid hex"))]
>     InvalidSecretKeyHex,

If yes, we need it as an opaque hex error type (the error from the `hex` crate can contain a character from the parsed string).

Reply:
I'm referring to:

```rs
    DecodingError(#[cfg_attr(feature = "std", from)] hex::FromHexError),
```

Reply:
Ah `DecodingError` is still used when decoding signatures from hex. It's ok to propagate the hex error there, since signatures are not secret.